src/api.c

```diff
@@ -3628,8 +3667,8 @@ int cgroup_get_cgroup(struct cgroup *cgroup)
 		} else { /* cgroup v2 */
 			bool enabled;
 
-			error = cgroupv2_get_subtree_control(path, cg_mount_table[i].name,
-							     &enabled);
+			error = cgroupv2_get_controllers(path, cg_mount_table[i].name,
+							 &enabled);
```

Comment:
It should be `cgroupv2_get_subtree_control()`? or else the `cg.get()` will return `cgroup.controllers` controllers.
```
>>> cg = Cgroup("a", Version.CGROUP_V2)
>>> cg.get()
>>> len(cg.controllers)
5
```
replacing `cgroupv2_get_controllers()` with `cgroupv2_get_subtree_control()`:
```
>>> from libcgroup import Cgroup, Version
>>> cg = Cgroup("a", Version.CGROUP_V2)
>>> cg.get()
>>> len(cg.controllers)
1
```

Reply:
My position is that reading the `subtree_control` file is incorrect in this case.  The subtree control file tells us what controllers are enabled for the _children_ cgroups, but it has no bearing on the controllers enabled in _this_ cgroup.  To get the controllers enabled in this cgroup we should be reading the `cgroup.controllers` file.

Reply:
Sorry, I missed a detail in the comment, I had enabled `cpu` controller using `# echo +cpu > /sys/fs/cgroup/a/cgroup.subtree_control`.  I am a little confused, I thought `cgroup.controllers` provides the list of controllers that can be enabled for the cgroup vs `cgroup.subtree_control` is the list of controllers enabled from the available controllers listed in `cgroup.controllers`.

Reply:
Or here's another way to think about it:
```
# mkdir -p /sys/fs/cgroup/foo/bar
# ls /sys/fs/cgroup/foo/bar/ | wc -l
15

### At this point, there are no enabled controllers in foo/bar

# echo +cpu > /sys/fs/cgroup/foo/cgroup.subtree_control
# echo +memory > /sys/fs/cgroup/foo/cgroup.subtree_control
# ls /sys/fs/cgroup/foo/bar/ | wc -l
36
# cat /sys/fs/cgroup/foo/bar/cgroup.subtree_control

### Now there are a whole bunch more files in foo/bar because the cpu
### and memory controllers have been enabled within bar.   And foo/bar's
### cgroup.subtree_control is empty, so if we were to make foo/bar/baz,
### it would have no activated controllers

# mkdir /sys/fs/cgroup/foo/bar/baz
# ls /sys/fs/cgroup/foo/bar/baz | wc -l
15
```

Reply:
The answer is way more complicated than one may expect, when a cgroup is created using `mkdir`, they need to know what resources/controllers the tasks in that cgroup will be restricted too, defining the characteristics of the newly created cgroup. Hence, `cgroup.controllers` file will always be an intersection of `cgroup->parent.subtree_control` file, defining the controllers available for the cgroup. One may argue, If that the case why are the knobs of the controller those are not enabled by writing into `cgroup.subtree_control` is visible along with the `cgroup.*` files such `cgroup.procs`, and they should be only visible when the controller is enabled? To answer that, we need to look at the other constraint that cgroup v2 imposes, which is only the leaf node without
any controller enabled should have the tasks. This can be better illustrated
with a diagram:
```
                  /
                c1
                 |
               c2
               /  \
             t1 t2
```

before we dive into the example, the crux of the cgroup in one-liner is "what controller resource is restricted vs unrestricted", all the resources are unrestricted until the controller is enabled in the parent cgroup or we explicitly
turn on some controller settings in the cgroup the tasks are running.

With the above example, `c1` is the parent, that decides which all controller resources shall be restricted to the task `t1` and `t2` by enabling and disabling the controller in its `cgroup.subtree_control` file.  Coming to our issue here, when we try to find the controllers enabled vs available or in other words unrestricted vs restricted to the tasks or children cgroups, we should be reading the `cgroup.subtree_control` of the parent or read `cgroup.controllers` file instead to find the controller who is restricting/enabled. Hence `cgroupv2_get_controllers()` seems to be the right option to go with.